### PR TITLE
Detect possible commit hashes in the commit messages and convert them to hyperlinks

### PR DESCRIPTION
**Pull request by mnaoumov**
Implements #2714

### CODE REVIEW

File: ResourceManager/CommitInformation.cs [hunk 1]
```diff
@@ -49,6 +50,8 @@ public static CommitInformation GetCommitInfo(CommitData data, bool showRevision
             string header = data.GetHeader(showRevisionsAsLinks);
             string body = "\n" + WebUtility.HtmlEncode(data.Body.Trim());
 
+            if (showRevisionsAsLinks)
+                body = GitRevision.Sha1HashShortRegex.Replace(body, m => LinkFactory.CreateCommitLink(m.Value));
```
Comment (Radrik5) [inline ResourceManager/CommitInformation.cs]: Will it replace full 40 char hash with the first 10 chars?

**Comment by Radrik5** [comment]
Please consider using 'git rev-parse --verify hash^{commit}' to ignore false positive matches.

File: ResourceManager/CommitInformation.cs (hunk 1 above)
Comment (mnaoumov) [inline ResourceManager/CommitInformation.cs]: Thanks @Radrik5 I fixed that

File: ResourceManager/CommitInformation.cs [hunk 2]
```diff
@@ -41,15 +41,27 @@ public static CommitInformation GetCommitInfo(GitModule module, string sha1)
...
+            if (showRevisionsAsLinks)
+                body = GitRevision.Sha1HashShortRegex.Replace(body, match => ProcessHashCandidate(module, hash: match.Value));
             return new CommitInformation(header, body);
         }
+
+        private static string ProcessHashCandidate(GitModule module, string hash)
+        {
+            if (module == null)
+                return hash;
+            string revParseResult = module.RunGitCmd(string.Format("rev-parse --verify --quiet {0}^{{commit}}", hash));
```
Comment (jbialobr) [inline ResourceManager/CommitInformation.cs]: Checking if hash is valid should be implemented in GitModule.

**Pull request merged**